tests/cephadm/test_cephadm_pull.py

```diff
@@ -0,0 +1,59 @@
+import requests
+import yaml
+
+from ceph.ceph_admin import CephAdmin
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class CephadmpullFailure(Exception):
+    pass
+
+
+def _get_latest_ceph_version(url):
+    """
+    Get latest ceph version from URL
+    Args:
+        url: Ceph container yaml link
+    Returns:
+        Latest ceph version
+    """
+
+    response = requests.get(url)
+    yaml_response = yaml.safe_load(response.text)
+    log.info(yaml_response)
+    ceph_versions = list(yaml_response.keys())
+    ceph_versions.sort()
+    return yaml_response[ceph_versions[-1]]["ceph-version"]
+
+
+def run(ceph_cluster, **kw):
+    """
+    Verify ceph version using cephadm pull option
+    Args:
+        ceph_cluster: Ceph cluster object
+        **kw :     Key/value pairs of configuration information to be used in the test.
+
+    Returns:
+        int - 0 when the execution is successful.
+    """
+
+    config = kw.get("config")
+    # Executing cephadm pull command
+    instance = CephAdmin(cluster=ceph_cluster, **config)
+    # Geting ceph version detail using cephadm pull command
+    cmd = "cephadm pull"
+    out, _ = instance.installer.exec_command(cmd, sudo=True)
+    yaml_response = yaml.safe_load(out)
+    ceph_version_cephadm = yaml_response["ceph_version"].split(" ")[2][:-6]
+    log.info(f"cephadm pull ceph version: '{ceph_version_cephadm}'")
+    # Geting ceph version detail from magna002 latest ceph container yaml
+    url = config["url"]
```

Comment:
Raise exception in case `url` is not passed through config

Reply:
Updated